src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -3,192 +3,215 @@
+        public DeploymentOptionProperty<bool> IgnoreDmlTriggerState { get; set; }
 
-        public bool IgnoreDmlTriggerOrder { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreDmlTriggerOrder { get; set; }
 
-        public bool IgnoreDefaultSchema { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreDefaultSchema { get; set; }
 
-        public bool IgnoreDdlTriggerState { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreDdlTriggerState { get; set; }
 
-        public bool IgnoreDdlTriggerOrder { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreDdlTriggerOrder { get; set; }
 
-        public bool IgnoreCryptographicProviderFilePath { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreCryptographicProviderFilePath { get; set; }
 
-        public bool VerifyDeployment { get; set; }
+        public DeploymentOptionProperty<bool> VerifyDeployment { get; set; }
 
-        public bool IgnoreComments { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreComments { get; set; }
 
-        public bool IgnoreColumnCollation { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreColumnCollation { get; set; }
 
-        public bool IgnoreAuthorizer { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreAuthorizer { get; set; }
 
-        public bool IgnoreAnsiNulls { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreAnsiNulls { get; set; }
 
-        public bool GenerateSmartDefaults { get; set; }
+        public DeploymentOptionProperty<bool> GenerateSmartDefaults { get; set; }
 
-        public bool DropStatisticsNotInSource { get; set; }
+        public DeploymentOptionProperty<bool> DropStatisticsNotInSource { get; set; }
 
-        public bool DropRoleMembersNotInSource { get; set; }
+        public DeploymentOptionProperty<bool> DropRoleMembersNotInSource { get; set; }
 
-        public bool DropPermissionsNotInSource { get; set; }
+        public DeploymentOptionProperty<bool> DropPermissionsNotInSource { get; set; }
 
-        public bool DropObjectsNotInSource { get; set; }
+        public DeploymentOptionProperty<bool> DropObjectsNotInSource { get; set; }
 
-        public bool IgnoreColumnOrder { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreColumnOrder { get; set; }
 
-        public bool IgnoreTablePartitionOptions { get; set; } // DW Specific
+        public DeploymentOptionProperty<bool> IgnoreTablePartitionOptions { get; set; } // DW Specific
 
-        public string AdditionalDeploymentContributorPaths { get; set; } = string.Empty;
+        public DeploymentOptionProperty<string> AdditionalDeploymentContributorPaths { get; set; }
 
-        public ObjectType[] DoNotDropObjectTypes { get; set; } = null;
+        public DeploymentOptionProperty<ObjectType[]> DoNotDropObjectTypes { get; set; } = new DeploymentOptionProperty<ObjectType[]>(null);
```

Comment:
Why null? Seems like you should just not define or have an empty array if there aren't any ObjectTypes you want to include here